pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -475,3 +534,20 @@ func (s *ClusterSpec) Expiration() time.Time {
 	}
 	return timeutil.Now().Add(l)
 }
+
+// TotalCPUs is the total amount of CPUs allocated for a cluster.
+func (s *ClusterSpec) TotalCPUs() int {
+	cpu := s.NodeCount * s.CPUs
+	crdbCPU := (s.NodeCount - 1) * s.CPUs
+	// If the test has a workload node, it may require different CPU than the rest of the cluster.
+	if s.WorkloadNode {
+		if s.WorkloadNodeCPUs != 0 {
+			cpu = crdbCPU + s.WorkloadNodeCPUs
+		} else if DefaultWorkloadNodeCPUs < s.CPUs && (s.NodeCount-1) <= SmallerWorkloadNodeThreshold {
```

Comment:
Normally derived fields would be ok, but here we have a coupling between this func and `RoachprodOpts`. The comment in the latter is likely to be ignored, i.e., we can't rely on comments to assert (correctness) invariants. 

One way to remove this coupling is to ensure that `s.WorkloadNodeCPUs` is always set; e.g., in `MakeClusterSpec` we set `s.CPUs` to 4 unless it's explicitly overridden.